plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -364,7 +364,7 @@ else if (type instanceof ObjectType) {
             ImmutableList.Builder<RowDecoder.NameAndDescriptor> decoderFields = ImmutableList.builder();
             for (IndexMetadata.Field rowField : objectType.getFields()) {
                 String name = rowField.getName();
-                TypeAndDecoder child = toTrino(appendPath(path, name), rowField);
```

Comment:
Does it fail only for timestamp or even for other types ? Do we have tests for other data types ?

Reply:
Other types look fine as far as I confirmed. Basic test cases exist in `testDataTypesNested`. 
https://github.com/trinodb/trino/blob/c4711c10e66d21ecce78a5fa261da8783ed0d1cb/plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java#L1549